chichiwang/exercise-js
# Implement jest as a test runner

## Description
Linked to issue #2
Implement [jest](https://jestjs.io/) into the project and run via the npm script `test`. A test suite is created to test the existing code `server/index.mjs`. Coverage is generated for the test runs and a minimum coverage of 100% is set.

## Changes
* `jest`, `@babel/core`, and `@babel/preset-env` are added as devDependencies
* npm script `test` is updated to run `jest`
* `babel.config.js` is created
  * Configure `@babel/preset-env` to target the current node version
* `jest.config.json` is created
  * Ensure `.mjs` files are transformed by `babel-jest` for test runs
  * Configure to collect coverage in `__coverage__/` directory
  * Configure to require 100% coverage
* `server/__tests__/index.test.js` is created to test `server/index.mjs`

## Steps to QA
* Pull down and switch to this branch
* Run `npm test`
  * Ensure tests pass
  * Ensure coverage is generated in `__coverage__/` directory
* Delete a couple tests
  * Run `npm test` and ensure that the process fails due to coverage despite all tests passing
